src/index.ts

```diff
@@ -199,65 +199,3 @@ function _decode(input: Buffer): Decoded {
     }
   }
 }
-
-/** Check if a string is prefixed by 0x */
-function isHexPrefixed(str: string): boolean {
-  return str.slice(0, 2) === '0x'
-}
-
-/** Removes 0x from a given String */
-function stripHexPrefix(str: string): string {
-  if (typeof str !== 'string') {
-    return str
-  }
-  return isHexPrefixed(str) ? str.slice(2) : str
-}
-
-/** Transform an integer into its hexadecimal value */
-function intToHex(integer: number | bigint): string {
-  if (integer < 0) {
-    throw new Error('Invalid integer as argument, must be unsigned!')
-  }
-  const hex = integer.toString(16)
-  return hex.length % 2 ? `0${hex}` : hex
-}
-
-/** Pad a string to be even */
-function padToEven(a: string): string {
-  return a.length % 2 ? `0${a}` : a
-}
-
-/** Transform an integer into a Buffer */
-function intToBuffer(integer: number | bigint): Buffer {
-  const hex = intToHex(integer)
-  return Buffer.from(hex, 'hex')
-}
-
-/** Transform anything into a Buffer */
-function toBuffer(v: Input): Buffer {
-  if (!Buffer.isBuffer(v)) {
-    if (typeof v === 'string') {
-      if (isHexPrefixed(v)) {
-        return Buffer.from(padToEven(stripHexPrefix(v)), 'hex')
-      } else {
-        return Buffer.from(v)
```

Comment:
Hmmm this might unfortunately be a breaking change because this version of toBuffer doesn't have the strictness that our latest util toBuffer has regarding 0x-prefixed strings.

>   1) RLP encoding (string):
       should return itself if single byte and less than 0x7f::
     Error: Cannot convert string to buffer. toBuffer only supports 0x-prefixed hex strings and this string was given: a

Reply:
maybe we save this PR for a v3 release, I don't think there is anything else planned for a next breaking change yet though, although one on the backburner that could be good to include would be #95

Reply:
might be able to work around this for now by just adding 0x prefix to strings without it in encode, and then disallow it again  in v3 for improved validation, I remember @alcuadrado saying there can be weird bugs when just accepting strings and converting to hex when they don't explicitly have the hex prefix.